src/plugins/lua/clickhouse.lua

```diff
@@ -988,6 +988,10 @@ local function get_last_removal_ago()
     f:close()
   end
 
+  if last_ts == nil or (last_ts + settings.retention.period) <= current_ts then
```

Comment:
I would suggest to use `if type(last_ts) ~= 'number' ... ` as it covers more potential misuse problems.

Reply:
`last_ts = tonumber(f:read('*number'))` - `tonumber` should return `number` or `nil`, there should be no other types.